### PR TITLE
Use AutoProperties for different Attributes and Extensions

### PR DESCRIPTION
**Pull request by thomasclaudiushuber**
Cleaning up attributes and extensions to use auto properties

### CODE REVIEW

File: src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Windows/Markup/ContentWrapperAttribute.cs [hunk 1]
```diff
@@ -34,47 +34,37 @@ sealed public class ContentWrapperAttribute : Attribute
...
+        public override object TypeId=> this; 
 
         /// <summary>
         ///     Overrides Object.Equals to implement correct equality semantics for this
         ///     attribute.
         /// </summary>
         public override bool Equals(object obj) 
         {
-            ContentWrapperAttribute other = obj as ContentWrapperAttribute;
+            var other = obj as ContentWrapperAttribute;
```
[the remark this answers is not in the document]
Comment (thomasclaudiushuber) [reply src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Windows/Markup/ContentWrapperAttribute.cs]: Yes, I agree. Personally I've made the experience in more and more projects that if there's a casting or a constructor call on the right side, then var makes sense on the left side. But I agree, it's personal preference.

File: src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Windows/Markup/DependsOnAttribute.cs [hunk 2]
```diff
@@ -41,27 +40,20 @@ public sealed class DependsOnAttribute : Attribute
...
         /// <summary>
         /// Override of Attribute.TypeId
         /// </summary>
         public override object TypeId
-        {
             // this allows multiple DependsOn attributes to show up in the
             // PropertyDescriptor.Attributes collection.
-            get { return this; }
-        }
+            => this;
```
[the remark this answers is not in the document]
Comment (thomasclaudiushuber) [reply src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Windows/Markup/DependsOnAttribute.cs]: I agree. The reason I didn't make a one-line was the two-slash comment on the getter and the three-slash-comment on the property itself. I think in this case it's better to stay with the property, or respectively use an expression for the getter.

[the remark this answers is not in the document]
Comment (thomasclaudiushuber) [reply src/Microsoft.DotNet.Wpf/src/System.Xaml/System/Windows/Markup/DependsOnAttribute.cs]: Ok, I pushed with a getter and an expression. It's cleaner with the comment on top of it.

**Comment by ericstj** [comment]
I typically rebase, which is our guidance: https://github.com/dotnet/corefx/wiki/git-reference#logical-workflow

If you do have a messy conflict to address in a rebase you can call that out in a github comment after pushing to recommend that reviewers take another look.  If its a simple fix to the conflict than its generally OK to assume it doesn't reset sign-off and merge once CI is green.

**Comment by thomasclaudiushuber** [comment]
Thanks @ericst & @karelz, helps a lot. The link is great.

It's actually the way I worked in the past on internal customer projects with git. Create feature-branch from master, implement feature, rebase on master, create PR, merge to master. But I was not sure if rebasing a published PR is a good idea. Thanks.

**Comment by rladuca** [comment]
@thomasclaudiushuber Thanks for the contribution and thanks to everyone for the discussion on this.  Merging.

**Pull request merged**